packages/block/src/types.ts

```diff
@@ -42,6 +42,13 @@ export interface BlockOptions {
    * pointing to a Shanghai block: this will lead to set the HF as Shanghai and not the Merge).
    */
   hardforkByTTD?: BigIntLike
+  /**
+   * Infer the total difficulty by inspecting the difficulty field and determine the HF.
+   *
+   * This option is used specially when we cannnot calculate the total difficulty
+   * for e.g. in the blocks constructed in skeleton/beacon sync
+   */
+  hardforkByChainTTD?: BigIntLike
```

Comment:
I would really not want to side introduce yet another `hardforkBy*` option here. This is such a special case and we already have two options here.

Remember, we are *not* developing these libraries for ourselves respectively - at least regarding the priority - these are *not* libraries for the client or something but primarily stand-alone libraries which are used by community developers for various different purposes (so there also shouldn't any client construct/internals mentions in the code docs e.g., like "block in skeleton/beacon sync").

Can you please go with another solution here, e.g. just create a respective Common to be passed in manually in the applying client code? Guess that should be possible.

Reply:
ummmm i actually want to "remove" the hardforkBy options in a followup PR where properly set common needs to be passed from "outside" itself as the only purpose of these options is to set the hardfork. just didn't want to intermix those changes here.

Reply:
You can also give this a bit of extra time, things are also getting dense on my side regarding the releases since so many things side stepped in today that I think I'll rather not do tomorrow any more but postpone to Monday morning. So you would have tomorrow as well for an update, then we can hopefully still take this in (and otherwise it would also be possible to do the client release 2-3 days later since this technically not necessarily needs to be published in sync with the other releases).

Reply:
Ah, sorry, changes like these are not possible any more anyhow. Are you aware that we are planning breaking release release candidates tommorow (or now eventually on monday)? These kind of changes would be breaking and after tomorrow there is a development stop on breaking changes.

Generally: can you *please* discuss these kind of ideas on issues before you just go for it? It would really be good if you give people a chance to give some opinion and open some room for discussion.